pkg/loop/internal/core/services/capability/capabilities_registry.go

```diff
@@ -264,6 +264,37 @@ func (cr *capabilitiesRegistryClient) Add(ctx context.Context, c capabilities.Ba
 	return nil
 }
 
+func (cr *capabilitiesRegistryClient) Remove(ctx context.Context, c capabilities.BaseCapability) error {
+	info, err := c.Info(ctx)
+	if err != nil {
+		return err
+	}
+
+	// Check the capability and the CapabilityType match here as the ServeNew method does not return an error
+	err = validateCapability(c, info.CapabilityType)
+	if err != nil {
+		return err
+	}
+
+	var cRes net.Resource
+	id, cRes, err := cr.ServeNew(info.ID, func(s *grpc.Server) {
```

Comment:
@justinkaseman Hmmmm... this is confusing -- why do we need to set up a server here?

I would have expected that calling info followed by remove would be enough.